py/dml/ctree.py

```diff
@@ -595,6 +596,22 @@ def mkInlinedMethod(site, method, pre, body, post):
             collapsed.append(stmt)
     return InlinedMethod(site, method, pre, collapsed, post)
 
+class IndicesAssert(Statement):
+    @auto_init
+    def __init__(self, site, method, indices): pass
+
+    def toc_stmt(self):
+        self.linemark()
+        length = len(self.indices)
+        indices = ', '.join(f'({idx.read()})' for idx in self.indices)
+        dimsizes = ', '.join(map(str, self.method.dimsizes))
+        out('DML_ASSERT_INDICES('
+            + f'_id_infos[{self.method.parent.uniq - 1}].logname, '
+            + f'"{self.method.name}", ((const uint32 []) {{ {indices} }}), '
+            + f'((const uint32 []) {{ {dimsizes} }}), {length});\n')
```

Comment:
I don't remember details, but IIRC, compound literals *sometimes* induce GCC slowness (I think this is a main source of vtable init slowness). IIRC it's ok as long as the total size is small, but I don't remember. Deserves profiling, since this affects all methods.

Reply:
benching reveals the use of compound literals make no difference vs. array variables.

Reply:
Though the loop _is_ relevant and slow. Tweaking eliminates the overhead.